frontend/src/components/Discover_Event/EventList.tsx

```diff
@@ -1,31 +1,27 @@
 import EventCard from "../shared/EventCard";
 import "./EventList.css";
-import { useNavigate } from "react-router-dom";
-// import { TicketType } from "@/types/ticket";
 import { EventType } from "@/types/event";
+import { useFilterParams } from "@/services/queries";
+import { useNavigate } from "react-router-dom";
 
-type EventListType = EventType[];
-
-const EventdataList = ({
-  eventDataFromParent,
-}: {
-  eventDataFromParent: EventListType;
-}) => {
+const EventdataList = ({ paramsFromParent }: { paramsFromParent: string }) => {
   const navigate = useNavigate();
-  // const displayMode = "grid";
-  const eventdatas = eventDataFromParent;
-  // const gridStyle = {
-  //   display: displayMode === "grid" ? "grid" : "block",
-  //   gridTemplateColumns: displayMode === "grid" ? "repeat(3, 1fr)" : "none",
-  //   gap: displayMode === "grid" ? "20px" : "0",
-  //   paddingLeft: "50px",
-  //   paddingRight: "50px",
-  // };
+  let params = paramsFromParent;
+  console.log(params);
+  const { data, error, isLoading } = useFilterParams(params || "");
+  console.log(data);
+  if (isLoading) return <div>Loading...</div>;
+  if (error)
+    return (
+      <div>Error occurred when trying to fetch data: {String(error)} </div>
+    );
+  if (data.message != undefined)
+    return <div>No data found for this Filter</div>;
+
+  const eventdatas = data;
 
-  const handleClick = (eventData: unknown) => {
-    navigate("/discover-event-details", {
-      state: eventData,
-    });
+  const handleClick = (eventData: EventType) => {
+    navigate(`/discover/${eventData.title}/event/${eventData._id}`);
```

Comment:
format the eventData.title so that space is replaced with -, so for instance if event name is 

Spring Time Coming should be /discover/spring-time-coming/event/58917295712895712

Reply:
`eventData.title` reformated
<img width="536" alt="image" src="https://github.com/Thangnguyenvn98/GatheringGlobe/assets/129610318/a061e10c-3d52-4994-a93b-7e08309d776b">
<img width="814" alt="image" src="https://github.com/Thangnguyenvn98/GatheringGlobe/assets/129610318/9ddc8d2e-8804-4a4c-8a56-43014a339b8a">